runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -2435,24 +2457,7 @@ J9::ValuePropagation::constrainRecognizedMethod(TR::Node *node)
                   return;
 
                TR_PrexArgInfo *argInfo = new (trStackMemory()) TR_PrexArgInfo(node->getNumChildren(), trMemory());
```

Comment:
This case and the new case are identical from this point on. I think that it would make sense to move at least the creation of the `TR_PrexArgInfo` into the new function. Adding the new entry to `_refinedMethodHandleINLMethodsToInline` could happen there as well if it were made a method of `J9::ValuePropagation`